### PR TITLE
Remove remaining references to GPU pragma/PGI compiler requirements

### PR DESCRIPTION
**Pull request by maxpkatz**
## PR checklist

- [x] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

File: Exec/Make.Castro [hunk 1]
```diff
@@ -51,26 +51,6 @@ endif
...
-# version of gcc on the system.
-
-ifeq ($(which_computer),summit)
-  ifeq ($(USE_CUDA),TRUE)
-    SUMMIT_GCC_OVERRIDE_DIR := /sw/summit/gcc/7.4.0
-    NVCC_CCBIN := $(SUMMIT_GCC_OVERRIDE_DIR)/bin/g++
-    LIBRARY_LOCATIONS += $(SUMMIT_GCC_OVERRIDE_DIR)/lib64
-    LIBRARIES += -Wl,-rpath,$(SUMMIT_GCC_OVERRIDE_DIR)/lib64
-  endif
-endif
```
Comment (zingale) [inline Exec/Make.Castro:72]: does this break Summit compilation with PGI now?

Comment (maxpkatz) [reply Exec/Make.Castro:72]: Yes.

Comment (maxpkatz) [reply Exec/Make.Castro:72]: Updated the workflow docs to indicate that compilation with gcc is now recommended.

**Comment by zingale** [comment]
all tests pass: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-12-16-003/index.html

**Review by zingale — APPROVED** [review]

**Pull request merged**